KeeTrayTOTP/TrayTOTP_Plugin.cs

```diff
@@ -206,12 +206,29 @@ internal IEnumerable<PwEntry> GetVisibleAndValidPasswordEntries()
         /// <summary>
         /// Get all the password entries in all groups and filter entries that are expired or have invalid TOTP settings.
         /// </summary>
-        /// <returns></returns>
         internal IEnumerable<PwEntry> GetVisibleAndValidPasswordEntries(PwGroup pwGroup)
         {
             var entries = pwGroup.GetEntries(true);
+                        
+            return entries.Where(entry => !entry.IsExpired() && TOTPEntryValidator.HasSeed(entry) && !InRecycleBin(entry));
+        }
+
+        /// <summary>
+        /// Returns true when there is a recycle bin, and the entry is inside of it.
+        /// </summary>
+        /// <param name="entry"></param>
+        private bool InRecycleBin(PwEntry entry)
+        {
+            if (PluginHost.Database.RecycleBinEnabled)
+            {
+                var pgRecycleBin = PluginHost.Database.RootGroup.FindGroup(PluginHost.Database.RecycleBinUuid, true);
+                if (pgRecycleBin != null)
```

Comment:
This is not working for multiple open databases, because ``GetVisibleAndValidPasswordEntries`` will be called for ``PwGroup``s in other databases as well.

Reply:
Good catch! Let me take a look.

Reply:
Great, thanks!

Could be a performance issue for big databases, as we are calling FindGroup for every single TOTP enabled entry.